Project.toml

```diff
@@ -1,16 +1,16 @@
 name = "LinLogQuantization"
 uuid = "44354dd6-1ce2-4e80-a0fb-98a3a2b05088"
 authors = ["Milan <milankloewer@gmx.de> and contributors"]
-version = "0.2.1"
+version = "1.0.0"
```

Comment:
The new notation requiring dims as keyword `dims = 4` is a breaking change, isn't it?

Reply:
@milankl I suggest the following: 

1. I reset the branch to a commit previous to the standardisation of the `dims` argument.
2. You release 0.3.0 with the new support for signed integers.
3. I create a following PR with the standardisation of the `dims` argument.
4. You release 1.0.0 with the standard notation.

Reply:
I like the idea of releasing something non breaking (but that would be 0.2.2!) but if you want to also go straight to v1.0 I could be supportive of that even though I don't see the point. But given that you've put more effort than me into this repo I'd be happy to include you as a collaborator and then you have write access too and can take the post v1 development and maintaince in the direction you like.

But could you elaborate how the `dims` keyword is breaking? It's an optional keyword no?

Reply:
> I like the idea of releasing something non breaking (but that would be 0.2.2!) 

mmm the most common versioning scheme is [Semantic Versioning (SemVer)](https://en.wikipedia.org/wiki/Software_versioning#Semantic_versioning), which uses a three-part number format: `MAJOR.MINOR.PATCH` . This is not a patch (a.k.a. bugfix), but anew functionality that doesn't break backwards compatibility, i.e. a _MINOR_. Am I missing something?

> But given that you've put more effort than me into this repo I'd be happy to include you as a collaborator and then you have write access too and can take the post v1 development and maintaince in the direction you like.

Becoming a collaborator would be great 😄 !

> But could you elaborate how the dims keyword is breaking? It's an optional keyword no?

Until now the signature of the function was 
```julia 
function LinQuantArray(::Type{TUInt},A::AbstractArray{T,N},dim::Int)
```
with `dim` being a positional argument. People could call the function in the two following ways: 
```julia-repl
julia> LinQuantArray(UInt8, A) # no dimension specified
``` 
or 
```julia-repl 
julia> LinQuantArray(UInt8, A, 4)
``` 

Using the standard **keyword** argument `dims`, the  signature 
```julia 
function LinQuantArray(
    ::Type{TInteger},
    A::AbstractArray{T,N};
    dims::Int,
    extrema::Option{Tuple} = nothing
)
```

People can call the function in the three following ways: 
```julia-repl
julia> LinQuantArray(UInt8, A) # no dimension specified
``` 
or
```julia-repl 
julia> LinQuantArray(UInt8, A, dims=4)
``` 
or
```julia-repl 
julia> LinQuantArray(UInt8, A; dims=4)
``` 

This means that previous calls will break and this is a breaking change.

Reply:
> > I like the idea of releasing something non breaking (but that would be 0.2.2!)
> 
> mmm the most common versioning scheme is [Semantic Versioning (SemVer)](https://en.wikipedia.org/wiki/Software_versioning#Semantic_versioning), which uses a three-part number format: `MAJOR.MINOR.PATCH`. The support for unsigned integers is not a patch (a.k.a. bugfix), but anew functionality that doesn't break backwards compatibility, i.e. a _MINOR_. Am I missing something?

This applies to v1.0 and later. Before v1 v0.1 -> v0.2 is considered a breaking change, and v0.1 -> v0.1.1 can be used for both bugfixes as well as new features. This is quite common because if there are other packages having a dependency on, say, v0.2 then v0.2.2 would be included in that compatibility. Otherwise people have to update their compat entries manually (or with compathelper) to v0.3 or v1.0. I'm happy to go to either of these to signal a new feature, but ideally not without also releasing v0.2.2 so that packages don't have to udpate.

Reply:
Oh alright. Didn't know that. 

Then we can release: 
* `0.2.2` with the support for unsigned integers
*  `1.0.0` with the standard `dims` and and update of the Julia version and the dependencies

I will update this PR to leave the `dims` change out and create a different PR for that.